src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -807,6 +821,9 @@ protected void ExecuteIndexing()
 
                     while (true)
                     {
+                        if (_indexDisabled)
```

Comment:
Maybe it would be better to stop the batch at the current point, not wait till it is completed?
Imagine a situation when we have 100M batch and we only processed 1M docs.

Disabling the index should IMO end the batch at 1M and commit the changes, not wait till other 99M is done. It can be done by ending the foreach loop in MapDocuments I think. Thoughts?

Reply:
it's being stopped here:
https://github.com/ravendb/ravendb/pull/4448/files#diff-a98877097408ea1a038c4101886f9a9bR2395

Reply:
Yes, my bad. Great!